libs/core/src/lib/slider/slider.component.ts

```diff
@@ -0,0 +1,761 @@
+    ViewEncapsulation
+} from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { DOWN_ARROW, LEFT_ARROW, RIGHT_ARROW, UP_ARROW } from '@angular/cdk/keycodes';
+
+import { fromEvent, Subject } from 'rxjs';
+import { debounceTime, takeUntil } from 'rxjs/operators';
+
+import { applyCssClass, CssClassBuilder, KeyUtil, RtlService } from '../utils/public_api';
+import { PopoverDirective } from '../popover/public_api';
+
+export const SLIDER_VALUE_ACCESSOR = {
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => SliderComponent),
+    multi: true
+};
+
+export enum SliderValueTargets {
+    SINGLE_SLIDER,
+    RANGE_SLIDER1,
+    RANGE_SLIDER2
+}
+
+export enum RangeHandles {
+    First,
+    Second
+}
+
+export interface SliderTickMark {
+    value: number | string;
+    label: string;
+    position?: number;
+}
+
+export type CustomValues = Omit<SliderTickMark, 'position'>;
+
+export type ControlValue = number | number[] | SliderTickMark | SliderTickMark[];
+
+const MIN_DISTANCE_BETWEEN_TICKS = 8;
+
+let sliderId = 0;
+
+@Component({
+    selector: 'fd-slider',
+    templateUrl: './slider.component.html',
+    styleUrls: ['./slider.component.scss'],
+    encapsulation: ViewEncapsulation.None,
+    changeDetection: ChangeDetectionStrategy.OnPush,
+    providers: [SLIDER_VALUE_ACCESSOR]
+})
+export class SliderComponent
+    implements OnInit, OnChanges, OnDestroy, AfterViewInit, ControlValueAccessor, CssClassBuilder {
+    /** Slider id, it has some default value if not set,  */
+    @Input()
+    @HostBinding('attr.id')
+    id = 'fd-slider-id-' + sliderId++;
+
+    /** User's custom classes */
+    @Input()
+    class: string;
+
+    /** Id of the element that labels object number. */
+    @Input()
+    ariaLabelledBy: string = null;
+
+    /** Aria label for the object number. */
+    @Input()
+    ariaLabel: string = null;
+
+    /** Get Minimum value. */
+    @Input()
+    get min(): number {
+        return this._min;
+    }
+
+    /** Set Minimum value. */
+    set min(value: number) {
+        if (((this.max - value) / this.step) % 1 !== 0) {
+            return;
+        }
+
+        this._min = value;
+    }
+
+    /** Get Maximum value. */
+    @Input()
+    get max(): number {
+        return this._max;
+    }
+
+    /** Set Maximum value. */
+    set max(value: number) {
+        if (((value - this.min) / this.step) % 1 !== 0) {
+            return;
+        }
+
+        this._max = value;
+    }
+
+    /** Get Step value. */
+    @Input()
+    get step(): number {
+        return this._step;
+    }
+
+    /** Set Step value. */
+    set step(value: number) {
+        if (((this.max - this.min) / value) % 1 !== 0) {
+            return;
+        }
+
+        this._step = value;
+    }
+
+    /** Jump value. */
+    @Input()
+    jump = 10;
+
+    /** Put a label on every N-th tickmark. */
+    @Input()
+    tickmarksBetweenLabels = 1;
+
+    /** Slider mode. */
```

Comment:
Please provide here the available options ('single'|'range') as this is shown to the user. Also add which is the default one.